### PR TITLE
docs: postinstall runs for any change

### PR DESCRIPTION
**Pull request by mrienstra**
As per https://github.com/yarnpkg/yarn/issues/8498#issuecomment-759572630

I also tested locally (`"packageManager": "yarn@3.0.2"`) and verified that `postinstall` runs in all of the following cases:
* yarn add (new)
* yarn remove
* yarn add (upgrade existing)
* yarn add (downgrade existing)
* remove manually from package.json then run `yarn`
* add manually to package.json then run `yarn`
* upgrade manually in package.json then run `yarn`
* downgrade manually in package.json then run `yarn`
* yarn up (upgrade)

Also pondering adding a note about a potential "gotcha" someone may encounter when attempting to debug `postinstall`: output from `echo` is not shown. When running the script below:

```json
"scripts": {
  "postinstall": "touch \"./postinstall_script_ran_$(date +\"%Y-%m-%d_%I-%M-%S-%p\")\" && echo 'postinstall script ran'"
},
```

... Running via `yarn postinstall` will echo "postinstall script ran" (and create a file), but when run as a side effect (e.g. `yarn add foo`), a file will be created but "postinstall script ran" is not shown (echo output).

### CODE REVIEW

File: packages/gatsby/content/advanced/lifecycle-scripts.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Packages can define in the `scripts` field of their manifest various actions tha
 
 - **prepublish** is called before `yarn npm publish` and similar commands (even before the package has been packed). This is the place where you'll want to check that the project is in an ok state. Because it's only called on prepublish, **the prepublish hook shouldn't have side effects.** In particular don't transpile the package sources in prepublish, as people consuming directly your repository (such as through the [`git:` protocol](/features/protocols#git)) wouldn't be able to use your project.
 
-- **postinstall** is called after a package got successfully installed on the disk. It is guaranteed to be called in topological order (in other words, your dependencies' postinstalls will always run before yours). For backwards compatibility, **preinstall** and **install** are called as part of **postinstall**.
+- **postinstall** is called whenever the dependency tree of your workspace changes -- e.g. when any package is added, removed or changed. It is guaranteed to be called in topological order (in other words, your dependencies' postinstalls will always run before yours). For backwards compatibility, **preinstall** and **install** are called as part of **postinstall**.
```
Comment (merceyz) [inline packages/gatsby/content/advanced/lifecycle-scripts.md:18]: It's not limited to workspaces, if the dependency tree of a `Package` changes its `postinstall` will be run anew

Comment (mrienstra) [reply packages/gatsby/content/advanced/lifecycle-scripts.md:18]: @merceyz: How about:
> is called whenever a dependency tree within the project changes -- e.g. when any package is added, removed or changed.

?

Comment (mrienstra) [reply packages/gatsby/content/advanced/lifecycle-scripts.md:18]: Or just:
>is called whenever a dependency changes -- e.g. when any package is added, removed or changed.

Comment (mrienstra) [reply packages/gatsby/content/advanced/lifecycle-scripts.md:18]: (Skimming https://yarnpkg.com/advanced/architecture for helpful language, but nothing is jumping out yet)

Comment (mrienstra) [reply packages/gatsby/content/advanced/lifecycle-scripts.md:18]: (Skimming https://yarnpkg.com/advanced/architecture for helpful language, but nothing is jumping out yet)

Comment (mrienstra) [reply packages/gatsby/content/advanced/lifecycle-scripts.md:18]: > is called after dependency tree changes are written to the disk -- e.g. after a package is added, removed or changed.

Based on https://yarnpkg.com/cli/install#details & https://yarnpkg.com/features/workspaces

**Comment by arcanis** [comment]
I've slightly updated this text to make it clearer that this only affects individual subtrees (ie if you depend on `foo` and `bar`, updating `bar` won't cause the `foo` postinstall to trigger anew because its own dependency tree will remain the same).

**Pull request merged**